TestHelper.psm1

```diff
@@ -201,3 +204,221 @@ function Install-ResourceDesigner {
+        run on.
+
+    .PARAMETER DSCResourceName
+        The full name of the DSC resource that the tests will be run on. This is
+        usually the name of the folder containing the actual resource MOF file.
+
+    .PARAMETER TestType
+        Specifies the type of tests that are being intialized. It can be:
+        Unit: Initialize for running Unit tests on a DSC resource. Default.
+        Integration: Initialize for running Integration tests on a DSC resource.
+
+    .OUTPUT
+        Returns a test environment object which must be passed to the
+        Restore-TestEnvironment function to allow it to restore the system
+        back to the original state as well as clean up and working/temp files.
+        
+    .EXAMPLE
+        $TestEnvrionment = Inialize-TestEnvironment `
+            -DSCModuleName 'xNetworking' `
+            -DSCResourceName 'MSFT_xFirewall'
+            
+        This command will initialize the test enviroment for Unit testing
+        the MSFT_xFirewall DSC resource in the xNetworking DSC module.      
+
+    .EXAMPLE
+        $TestEnvrionment = Inialize-TestEnvironment `
+            -DSCModuleName 'xNetworking' `
+            -DSCResourceName 'MSFT_xFirewall'
+            -TestType Integration
+            
+        This command will initialize the test enviroment for Integration testing
+        the MSFT_xFirewall DSC resource in the xNetworking DSC module.    
+#>
+function Initialize-TestEnvironment
+{
+    [OutputType([PSObject])]
+    [CmdletBinding()]
+    Param
+    (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String] $DSCModuleName,
+        
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String] $DSCResourceName,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateSet('Unit','Integration')]
+        [String] $TestType
+    )
+    
+    Write-Verbose -Verbose (`
+        'Initializing Test Environment for {0} testing of {1} in module {2}.' `
+            -f $TestType,$DSCResourceName,$DSCModuleName)   
+    if ($TestType -eq 'Unit')
+    {
+        [String] $RelativeModulePath = "DSCResources\$DSCResourceName\$DSCResourceName.psm1"
+    }
+    else
+    {
+        [String] $RelativeModulePath = "$DSCModuleName.psd1"
+    }
+
+    # Temp Working Folder - always gets removed on completion
+    # The tests can put anything in here and it will get cleaned up.
+    [String] $WorkingFolder = Join-Path -Path $env:Temp -ChildPath $DSCResourceName
+    
+    # Copy to Program Files for WMF 4.0 Compatability as it can only find resources in a few known places.
+    [String] $moduleRoot = Join-Path -Path "${env:ProgramFiles}\WindowsPowerShell\Modules" -ChildPath $DSCModuleName
+    
+    # If this module already exists in the Modules folder, make a copy of it in
+    # the temporary folder so that it isn't accidentally used in this test.
+    if(-not (Test-Path -Path $moduleRoot))
```

Comment:
I'm not sure we should be removing, even temporarily any modules which are already on user's machine.
I think the simple, non-invading solution would be to place module to be tested in some temporary location and add that location to the beginning of $env:PsModulePath until we finish testing.
What do you guys think?